### ISSUE TITLE
Pimcore Password encoder not working with Symfony ^5.3

### ISSUE THREAD

**Issue by dpfaffenbauer** [pimcore/pimcore#9489]
Symfony changed password encoders to password hashers in 5.3. They sort of also made it BC, but just sort-of.

I am having issues with the form constraint `Symfony\Component\Security\Core\Validator\Constraints\UserPassword`.

Howto reproduce:

 - Create a Frontend Authenticated User DataObject
 - Create a ChangePassword Form with the Constraint:

```php

declare(strict_types=1);

namespace CoreShop\Bundle\CustomerBundle\Form\Type;

use Symfony\Component\Form\AbstractType;
use Symfony\Component\Form\Extension\Core\Type\PasswordType;
use Symfony\Component\Form\Extension\Core\Type\RepeatedType;
use Symfony\Component\Form\FormBuilderInterface;
use Symfony\Component\Security\Core\Validator\Constraints\UserPassword;

class ChangePasswordType extends AbstractType
{
    public function buildForm(FormBuilderInterface $builder, array $options): void
    {
        $builder->add('current_password', PasswordType::class, [
            'label' => 'coreshop.form.customer.current_password',
            'mapped' => false,
            'constraints' => new UserPassword(),
        ]);

        $builder
            ->add('password', RepeatedType::class, [
                'type' => PasswordType::class,
                'invalid_message' => 'coreshop.form.customer.password.must_match',
                'first_options' => ['label' => 'coreshop.form.customer.new_password'],
                'second_options' => ['label' => 'coreshop.form.customer.new_password_repeat'],
            ]);
    }
}

```

Render and submit the form, you'll get this message:

```
No password hasher has been configured for account "Pimcore\Model\DataObject\CoreShopCustomer".
```

There is a way to fix this, but that requires different configurations and is not backward compatible to Symfony 5.2

**Comment by dvesh3** [comment]
@dpfaffenbauer does this fix your problem https://github.com/pimcore/pimcore/pull/9397? we had to override the service to make it backward compatible.

**Comment by dpfaffenbauer** [comment]
No, just tried it minutes before I posted the issue. The Password Constraint works a bit different. Things like login work though.

**Comment by dvesh3** [comment]
seems there's no quick fix, so need to spend more time on it.

**Comment by dpfaffenbauer** [comment]
sure... had a look as well and could not find a quick fix for it

**Comment by brusch** [comment]
Partly solved by #9611 
We'll close this as soon as 10.1 follow up PR #9612 is also fixed

**Comment by dpfaffenbauer** [comment]
perfect, thanks
